cmd/pj-rehearse/main.go

```diff
@@ -114,10 +119,45 @@ func main() {
 		gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
-	jobSpec, err := pjdwapi.ResolveSpecFromEnv()
-	if err != nil {
-		logrus.WithError(err).Error("could not read JOB_SPEC")
-		gracefulExit(o.noFail, misconfigurationOutput)
+	var jobSpec *pjdwapi.JobSpec
+	if o.local {
+		pullRef, err := git(o.candidatePath, []string{"rev-parse", "--abbrev-ref", "HEAD"})
+		if err != nil {
+			logrus.WithError(err).Error("could not get current branch")
+			gracefulExit(o.noFail, misconfigurationOutput)
+		}
+		baseRef, err := git(o.candidatePath, []string{"rev-parse", "--abbrev-ref", pullRef + "@{upstream}"})
```

Comment:
I imagine people will not always have dev workflows that work with this. I think all of Prow is well-formed to use the SHA whenever present and fall back to the ref if not, so we _could_ set these to dummy values or at least be best-effort about them. Let's see if people end up erroring on this before we change anything.

Reply:
I went with the idea of an upstream branch because it is a concept that is native to git, but I am not crazy about it and am open to suggestions. Some others I considered:

- `master`: why not?
- a separate flag: sounds like too much work.
- some other git magic I am no aware of?

Reply:
Let's use `master` if the user has not set an `upstream`? We don't have any other upstream branches on the `release` repo anyway right?